generator/test/good_test.dart

```diff
@@ -0,0 +1,264 @@
+import 'package:build_test/build_test.dart';
+import 'package:realm_generator/realm_generator.dart';
+import 'package:test/test.dart';
+
+void main() {
+  test('required argument with default value', () async {
+    await testBuilder(
+      generateRealmObjects(),
+      {
+        'pkg|lib/src/test.dart': r'''
+import 'package:realm_common/realm_common.dart';
+
+part 'test.g.dart';
+
+@RealmModel()
+class _Person {
+  int age = 47;
+}''',
+      },
+      outputs: {
+        'pkg|lib/src/test.realm_objects.g.part': '// **************************************************************************\n'
+            '// RealmObjectGenerator\n'
+            '// **************************************************************************\n'
+            '\n'
+            'class Person extends _Person with RealmObject {\n'
+            '  static var _defaultsSet = false;\n'
+            '\n'
+            '  Person({\n'
+            '    int age = 47,\n'
+            '  }) {\n'
+            '    _defaultsSet = _defaultsSet ||\n'
+            '        RealmObject.setDefaults<Person>({\n'
+            '          \'age\': 47,\n'
+            '        });\n'
+            '    this.age = age;\n'
+            '  }\n'
+            '\n'
+            '  Person._();\n'
+            '\n'
+            '  @override\n'
+            '  int get age => RealmObject.get<int>(this, \'age\') as int;\n'
+            '  @override\n'
+            '  set age(int value) => RealmObject.set(this, \'age\', value);\n'
+            '\n'
+            '  static SchemaObject get schema => _schema ??= _initSchema();\n'
+            '  static SchemaObject? _schema;\n'
+            '  static SchemaObject _initSchema() {\n'
+            '    RealmObject.registerFactory(Person._);\n'
+            '    return const SchemaObject(Person, [\n'
+            '      SchemaProperty(\'age\', RealmPropertyType.int),\n'
+            '    ]);\n'
+            '  }\n'
+            '}\n'
+            '',
+      },
+      reader: await PackageAssetReader.currentIsolate(),
+    );
+  });
+
+  test('required argument', () async {
+    await testBuilder(
+      generateRealmObjects(),
+      {
+        'pkg|lib/src/test.dart': r'''
+import 'package:realm_common/realm_common.dart';
+
+part 'test.g.dart';
+
+@RealmModel()
+class _Person {
+  @PrimaryKey()
+  late final String name;
+}''',
+      },
+      outputs: {
+        'pkg|lib/src/test.realm_objects.g.part': '// **************************************************************************\n'
+            '// RealmObjectGenerator\n'
+            '// **************************************************************************\n'
+            '\n'
+            'class Person extends _Person with RealmObject {\n'
+            '  static var _defaultsSet = false;\n'
+            '\n'
+            '  Person(\n'
+            '    String name,\n'
+            '  ) {\n'
+            '    _defaultsSet = _defaultsSet || RealmObject.setDefaults<Person>({});\n'
```

Comment:
not sure I my comment is still there so writing it here again. We don't need to generate the defaults handling code when there are not defaults

Reply:
Fixed